framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -2033,8 +2038,13 @@ AutomaticMortarGeneration::projectPrimaryNodesSinglePair(
               // Special case: xi1=+/-1.
               // We shouldn't get here, because this primary node should already
               // have been mapped during the project_secondary_nodes() routine.
-              throw MooseException("We should never get here, aligned primary nodes should already "
-                                   "have been mapped.");
+              // I think there is a chance we get here since the tolerances are applied to
+              // the xi coordinate and that value may be different on a primary element and a
+              // secondary element since they may have different sizes.
```

Comment:
Maybe something like

```
// It is unlikely that we get here, because this primary node should already
// have been mapped during the project_secondary_nodes() routine, but
// there is still a chance since the tolerances are applied to
// the xi coordinate and that value may be different on a primary element and a
// secondary element since they may have different sizes.
```